src/sql-parser/src/ast/defs/query.rs

```diff
@@ -331,6 +390,31 @@ impl<T: AstInfo> AstDisplay for Cte<T> {
 }
 impl_display_t!(Cte);
 
+use crate::ast::ColumnDef;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CteMutRec<T: AstInfo> {
+    pub name: Ident,
+    pub columns: Vec<ColumnDef<T>>,
+    pub id: T::CteId,
+    pub query: Query<T>,
+}
+
+impl<T: AstInfo> AstDisplay for CteMutRec<T> {
+    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
+        f.write_node(&self.name);
```

Comment:
Aren't we missing the `MUTUALLY RECURSIVE` bit?

```suggestion
        f.write_str("MUTUALLY RECURSIVE");
        f.write_node(&self.name);
```

Reply:
We should add some parser tests here! testdata directory in this crate. That'll let you test that the syntax roundtrips, even before the planner is all wired up.

Reply:
Maybe! I don't know what `AstDisplay` is for. It would put `MUTUALLY RECURSIVE` in front of each binding, none of which have `WITH` in front of them, so it's not like it produces the SQL that would have generated these. Perhaps `CteBlock` wants to put this instead if that is the goal?

Reply:
Yeah, right now (`main`) `Query` is the thing that blats down `WITH` and each `Cte` has nothing to say other than its binding. Perhaps The right thing is for `CteBlock` to determine 1. is there anything to print at all, and 2. should it be `WITH` or `WITH MUTUALLY RECURSIVE`.